contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,1212 @@
+        emit Delegated(operatorAddress, operatorAddress, delegation, delegation);
+        emit Delegated(operatorAddress, DEAD_ADDRESS, LOCK_AMOUNT, LOCK_AMOUNT);
+
+        IGovToken(GOV_TOKEN_ADDR).sync(creditContract, operatorAddress);
+    }
+
+    /**
+     * @param newConsensusAddress the new consensus address of the validator
+     */
+    function editConsensusAddress(address newConsensusAddress)
+        external
+        whenNotPaused
+        notInBlackList
+        validatorExist(msg.sender)
+    {
+        if (newConsensusAddress == address(0)) revert InvalidConsensusAddress();
+        if (consensusToOperator[newConsensusAddress] != address(0) || _legacyConsensusAddress[newConsensusAddress]) {
+            revert DuplicateConsensusAddress();
+        }
+
+        address operatorAddress = msg.sender;
+        Validator storage valInfo = _validators[operatorAddress];
+        if (valInfo.updateTime + BREATHE_BLOCK_INTERVAL > block.timestamp) revert UpdateTooFrequently();
+
+        consensusExpiration[valInfo.consensusAddress] = block.timestamp;
+        valInfo.consensusAddress = newConsensusAddress;
+        valInfo.updateTime = block.timestamp;
+        consensusToOperator[newConsensusAddress] = operatorAddress;
+
+        emit ConsensusAddressEdited(operatorAddress, newConsensusAddress);
+    }
+
+    /**
+     * @param commissionRate the new commission rate of the validator
+     */
+    function editCommissionRate(uint64 commissionRate)
+        external
+        whenNotPaused
+        notInBlackList
+        validatorExist(msg.sender)
+    {
+        address operatorAddress = msg.sender;
+        Validator storage valInfo = _validators[operatorAddress];
+        if (valInfo.updateTime + BREATHE_BLOCK_INTERVAL > block.timestamp) revert UpdateTooFrequently();
+
+        if (commissionRate > valInfo.commission.maxRate) revert InvalidCommission();
+        uint256 changeRate = commissionRate >= valInfo.commission.rate
+            ? commissionRate - valInfo.commission.rate
+            : valInfo.commission.rate - commissionRate;
+        if (changeRate > valInfo.commission.maxChangeRate) revert InvalidCommission();
+
+        valInfo.commission.rate = commissionRate;
+        valInfo.updateTime = block.timestamp;
+
+        emit CommissionRateEdited(operatorAddress, commissionRate);
+    }
+
+    /**
+     * @notice the moniker of the validator will be ignored as it is not editable
+     * @param description the new description of the validator
+     */
+    function editDescription(Description memory description)
+        external
+        whenNotPaused
+        notInBlackList
+        validatorExist(msg.sender)
+    {
+        address operatorAddress = msg.sender;
+        Validator storage valInfo = _validators[operatorAddress];
+        if (valInfo.updateTime + BREATHE_BLOCK_INTERVAL > block.timestamp) revert UpdateTooFrequently();
```

Comment:
`UpdateTooFrequently` check is used very frequently with the same code. Considering to make it a modifier.

Reply:
It needs `valInfo`. If move these code to modifer, the function will read the `valInfo` twice.